File: backend/app/models/proxy_server.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class ProxyServer(Base):
    __tablename__ = "proxy_servers"

    id: Mapped[int] = mapped_column(primary_key=True)
    host: Mapped[str] = mapped_column(String(255), nullable=False)
    port: Mapped[int] = mapped_column(Integer, nullable=False)
    protocol: Mapped[str] = mapped_column(String(32), default="vless", nullable=False)
    region: Mapped[str] = mapped_column(String(64), nullable=False)
    username: Mapped[str | None] = mapped_column(String(255), nullable=True)
    password: Mapped[str | None] = mapped_column(String(255), nullable=True)
    # VLESS+Reality (заполняются для protocol="vless")
    uuid: Mapped[str | None] = mapped_column(String(64), nullable=True)
    flow: Mapped[str | None] = mapped_column(String(32), nullable=True)
    public_key: Mapped[str | None] = mapped_column(String(128), nullable=True)
    short_id: Mapped[str | None] = mapped_column(String(32), nullable=True)
    server_name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    latency_ms: Mapped[int | None] = mapped_column(Integer, nullable=True)
    last_checked_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True), nullable=True
    )
    expires_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )

File: frontend_macos/SplitVPN/Models/ProxyServer.swift
import Foundation

/// Прокси-сервер, как его отдаёт бекенд (/proxy/list, /proxy/best).
struct ProxyServer: Codable, Identifiable, Equatable {
    let id: Int
    let host: String
    let port: Int
    let `protocol`: String
    let region: String
    let username: String?
    let password: String?
    let uuid: String?
    let flow: String?
    let publicKey: String?
    let shortId: String?
    let serverName: String?
    let isActive: Bool
    let latencyMs: Int?
    let expiresAt: Date?

    enum CodingKeys: String, CodingKey {
        case id, host, port, region, username, password, uuid, flow
        case `protocol`
        case publicKey = "public_key"
        case shortId = "short_id"
        case serverName = "server_name"
        case isActive = "is_active"
        case latencyMs = "latency_ms"
        case expiresAt = "expires_at"
    }

    var asProxyConfig: ProxyConfig {
        ProxyConfig(
            host: host,
            port: port,
            protocolType: `protocol`,
            username: username,
            password: password,
            region: region,
            uuid: uuid,
            flow: flow,
            publicKey: publicKey,
            shortId: shortId,
            serverName: serverName
        )
    }
}
